src/typesystem.hpp

```diff
@@ -30,9 +30,12 @@ struct TypeFunctionData {
 //
 // Dummy type functions are for unifying purposes only, but do not count
 // as 'deduced', because they were not created by the user
+
+using TypeFunctionStructure = std::unordered_map<std::string, MonoId>;
```

Comment:
Hold off on the typedefs. They make code less readable

Reply:
This is not always the case, but when there is a different name for every possible instanciation of `std::unordered_map` and `std::vector`, it hinders more than it helps